src/analyses/tmpSpecial.ml

```diff
@@ -13,13 +13,29 @@ struct
 
   let name () = "tmpSpecial"
   module ML = LibraryDesc.MathLifted
-  module Deps = SetDomain.Reverse (SetDomain.ToppedSet (CilType.Exp) (struct let topname = "All" end))
+  module Deps = SetDomain.ToppedSet (CilType.Exp) (struct let topname = "All" end)
   module MLDeps = Lattice.Prod (ML) (Deps)
-  module D = MapDomain.MapBot (Mval.Exp) (MLDeps)
+
+  (* x maps to (mathf(arg), deps) arg if x = mathf(arg) *)
+  module MM = MapDomain.MapBot_LiftTop (Mval.Exp) (MLDeps)
+
+  (* arg maps to (x, mathf(arg), deps) if x = mathf(arg) *)
+  module ExpFlat = Lattice.Flat (CilType.Exp) (Printable.DefaultNames)
+  module MInvEntry = Lattice.Prod3 (ExpFlat) (ML) (Deps)
```

Comment:
Have you tried this type:

~~~OCaml
module MInvEntry = Printable.Prod3 (CilType.Exp) (LibraryDesc.MathPrintable) (SetDomain.Make(CilType.Exp))
~~~

Might make it easier on the inside to give up on the lattice structure that is unused anyway.